public/js/services/chat.js

```diff
@@ -0,0 +1,102 @@
+angular.module('mean.system')
+  .factory('chat', ['env', function (env) {
+    /**
+    * Class to implement chat functionality
+    */
+    class Chat {
+      /**
+      * Constructor to create a new instance of this class
+      * @param{Object} dotEnv - Object holding our enviroment variables
+      * for angular
+      */
+      constructor(dotEnv) {
```

Comment:
You can just call this `env` though, so we don't confuse it with the dotenv node module